src/main/java/mate/controller/CarAddController.java

```diff
@@ -0,0 +1,54 @@
+package mate.controller;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.model.Manufacturer;
+import mate.service.CarService;
+import mate.service.ManufacturerService;
+
+@WebServlet(urlPatterns = "/cars/add")
+public class CarAddController extends HttpServlet {
+    private ManufacturerService manufacturerService;
+    private CarService carService;
+    private List<Manufacturer> manufacturers;
+
+    @Override
+    public void init() throws ServletException {
+        Injector injector = Injector.getInstance("mate");
+        manufacturerService =
+                (ManufacturerService) injector.getInstance(ManufacturerService.class);
+        carService = (CarService) injector.getInstance(CarService.class);
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        manufacturers = manufacturerService.getAll();
+        req.setAttribute("manufacturers", manufacturers);
+        req.getRequestDispatcher("/WEB-INF/views/cars/addCar.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Car car = new Car();
+        car.setModel(req.getParameter("model"));
+        Manufacturer manufacturer = manufacturers.stream()
```

Comment:
you should get id of manufacturer from post request and find a manufacturer by it

Reply:
Why? This case is very uncomfortable for user.